core/templates/domain/summary/exploration-summary-backend-api.service.spec.ts

```diff
@@ -15,96 +15,119 @@
-      var deferred = $q.defer();
-      deferred.resolve('sample-csrf-token');
-      return deferred.promise;
+import { HttpClientTestingModule, HttpTestingController } from
+  '@angular/common/http/testing';
+import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';
+
+import { AlertsService } from 'services/alerts.service';
+import { CsrfTokenService } from 'services/csrf-token.service.ts';
+import { ExplorationSummaryBackendApiService } from
+  'domain/summary/exploration-summary-backend-api.service.ts';
+
+
+describe('Exploration Summary Backend Api Service', () => {
+  let explorationSummaryBackendApiService:
+    ExplorationSummaryBackendApiService = null;
+  let httpTestingController: HttpTestingController;
+  let csrfService: CsrfTokenService = null;
+  let alertsService: AlertsService = null;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
     });
-  }));
 
-  afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
+    explorationSummaryBackendApiService = TestBed.get(
+      ExplorationSummaryBackendApiService
+    );
+    csrfService = TestBed.get(CsrfTokenService);
+    alertsService = TestBed.get(AlertsService);
+    httpTestingController = TestBed.get(HttpTestingController);
+
+    spyOn(csrfService, 'getTokenAsync').and.callFake(function() {
+      return Promise.resolve('sample-csrf-token');
+    });
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
   });
 
   it('should not load public exploration summaries from backend when' +
-    ' exploration id is not valid', function() {
-    var successHandler = jasmine.createSpy('success');
-    var failHandler = jasmine.createSpy('fail');
-    var explorationIds = ['#', null, '1'];
-    var alertSpy = spyOn(AlertsService, 'addWarning').and.callThrough();
+    ' exploration id is not valid', fakeAsync(()=> {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let explorationIds = ['#', null, '1'];
+    let alertSpy = spyOn(alertsService, 'addWarning').and.callThrough();
 
-    ExplorationSummaryBackendApiService.loadPublicExplorationSummaries(
+    explorationSummaryBackendApiService.loadPublicExplorationSummaries(
       explorationIds).then(successHandler, failHandler);
-    $rootScope.$apply();
+
+    flushMicrotasks();
 
     expect(alertSpy).toHaveBeenCalledWith(
       'Please enter a valid exploration ID.');
     expect(successHandler).not.toHaveBeenCalled();
     expect(failHandler).toHaveBeenCalledWith([null, null, null]);
-  });
+  }));
+
+  it('should use reject handler when server throws an error', fakeAsync(()=> {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let explorationIds = ['12'];
+    let requestUrl = '/explorationsummarieshandler/data?' +
```

Comment:
const, here and everywhere else applicable.

Reply:
done!